vault/auth.go

```diff
@@ -148,6 +155,63 @@ func (v *Vault) UserPassLogin() string {
 	return secret.Auth.ClientToken
 }
 
+// EC2Login - AWS EC2 auth backend
+func (v *Vault) EC2Login() string {
+	if env.Getenv("VAULT_AUTH_AWS_METHOD", "ec2") != "ec2" {
+		return ""
+	}
+
+	if skip := env.Getenv("VAULT_AUTH_AWS_EC2_SKIP", ""); skip != "" {
```

Comment:
I think if `VAULT_AUTH_AWS_METHOD` is required (i.e. _must_ be set to `ec2`, and doesn't default), then this environment variable can be dropped.